Ficek_QuantiApp/Ficek_QuantiApp/Domain/Tests/RocketClientTests/RocketClient+sadTests.swift

```diff
@@ -8,139 +8,206 @@ import RequestBuilder
-           
-        } operation: {
-            RocketClient.liveValue
-        }
-        
-       sut.fetchAllRockets()
-                .sink(
-                    receiveCompletion: { completion in
-                        switch completion {
-                        case .finished: break
-                        case let .failure(error):
-                            if error.underlyingError is NetworkError {
-                                XCTFail("Error shouldnt be networkError")
-                            } else {
-                                XCTFail("Shouldnt fail")
-                            }
-                        }
-                        expectation.fulfill()
-                        
-                    }, receiveValue: { rocket in
-                        XCTAssertNotEqual(rocket, [Rocket].mockTest)
-                    }
-                )
-                .store(in: &cancellables)
-
-            waitForExpectations(timeout: 10)
+    let mockUrlRequester = URLRequester { _ in
+      Just((Data(), urlResponse))
+        .setFailureType(to: URLError.self)
+        .eraseToAnyPublisher()
+    }
+    
+    let failedNetworkClient = NetworkClient(
+      urlRequester: mockUrlRequester,
+      networkMonitorClient: .live(onQueue: .main)
+    )
+    
+    let sut = withDependencies { dependency in
+      dependency.networkClient = failedNetworkClient
+    } operation: {
+      RocketClient.liveValue
+    }
+    
+    do {
+      let _ = try await sut.fetchAllRocketsAsync()
+    } catch is NetworkError {
+      errorRecieved = true
     }
+    
+    XCTAssertTrue(errorRecieved, "Error was not recieved")
+  }
+  
+  func test_fail_convertor_async() async throws {
+    let successResponse = try JSONEncoder().encode([RocketDTO].mockTest)
+    var errorRecieved = false
+    let mockResponse = HTTPURLResponse(
+      url: URL(string: "https://api.spacexdata.com/v4/rockets")!,
+      statusCode: 200,
+      httpVersion: nil,
+      headerFields: [
+        "\(HTTPHeaderName.acceptType.rawValue)": "\(AcceptTypeValue.json.rawValue)",
+        "\(HTTPHeaderName.contentType.rawValue)": "\(ContentTypeValue.json.rawValue)"
+      ]
+    )
+    
+    let mockUrlRequester = URLRequester { _ in
+      Just((successResponse, mockResponse!))
+        .setFailureType(to: URLError.self)
+        .eraseToAnyPublisher()
+    }
+    
+    let networkClient = NetworkClient(
+      urlRequester: mockUrlRequester,
+      networkMonitorClient: .live(onQueue: .main)
+    )
+    
+    let rocketsConverter = RocketsConverter(
+      externalModelConverter: { rocket in
+        return nil
+      }, domainModelConverter: { rocketDTO in
+        return nil
+      }
+    )
+    
+    let sut = withDependencies { dependency in
+      dependency.networkClient = networkClient
+      dependency.rocketsConverter = rocketsConverter
+    } operation: {
+      RocketClient.liveValue
+    }
+    
+    do {
+      let _ = try await sut.fetchAllRocketsAsync()
```

Comment:
Tady na to existuje urcite uz nejakej `XCTThrows` nebo neco takovyho, at tu nemusis delat ten do-catch explicitne.